receiver/dockerstatsreceiver/config.go

```diff
@@ -61,6 +67,26 @@ type Config struct {
 
 	// Metrics config. Enable or disable stats by name.
 	MetricsConfig metadata.MetricsSettings `mapstructure:"metrics"`
+
+	// Extract section allows specifying extraction rules to extract
+	// data from container specs
+	Extract ExtractConfig `mapstructure:"extract"`
+}
+
+// ExtractConfig section allows specifying extraction rules to extract
+// data from container specs.
+type ExtractConfig struct {
+	// EnvVars allows extracting data from container env vars and record it
+	// as resource attributes.
+	// It is a list of FieldExtractConfig type. See FieldExtractConfig
+	// documentation for more details.
+	EnvVars []container.FieldExtractConfig `mapstructure:"env_vars"`
+
+	// Labels allows extracting data from container labels and record it
+	// as resource attributes.
+	// It is a list of FieldExtractConfig type. See FieldExtractConfig
+	// documentation for more details.
+	Labels []container.FieldExtractConfig `mapstructure:"labels"`
 }
 
 func (config Config) Validate() error {
```

Comment:
Perhaps it is a good idea to log a warning if users are using the deprecated configuration.